scripts/install/jdownloader.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+# JDownloader Installer for swizzin
+# Author: Aethaeran
+
+# References
+# https://swizzin.ltd/dev/structure/
+# https://www.digitalocean.com/community/tutorials/how-to-install-java-with-apt-on-debian-10
+# https://support.jdownloader.org/Knowledgebase/Article/View/install-jdownloader-on-nas-and-embedded-devices
+# Liara already made a doc for installing JDownloader manually https://docs.swizzin.net/guides/jdownloader/
+# https://linuxize.com/post/how-to-check-if-string-contains-substring-in-bash/
+# https://linuxize.com/post/bash-check-if-file-exists/
+
+# TODO: Figure out if these imports are even necessary
+. /etc/swizzin/sources/globals.sh
+. /etc/swizzin/sources/functions/users
+
+# Functions
+function install_jdownloader() {
+
+    # Get my.jdownloader info for user
+    # TODO: Should double check to confirm everything is accurate, and loop back if anything isn't filled out.
+    # TODO: swizzin likely has utils for this already
+    echo_query "The user will require an account at https://my.jdownloader.org/ in order to access your JDownloader installation's web UI.\nIt is recommended to use a randomly generated password for the account since the password is save in plain text in the program folder.\nEnter the e-mail used to access this account once one is created:"
+    read -r 'myjd_email'
+    echo_query "Please enter the password for the account."
+    read -r 'myjd_password'
+    echo_query "Please enter the device name they would like this JDownloader installation to show up as."
+    read -r 'myjd_devicename'
+
+    # Install JDownloader
+    echo_progress_start "Downloading and installing JDownloader for $user"
+    mkdir -p /home/"$user"/jd
+    wget -q http://installer.jdownloader.org/JDownloader.jar -O /home/"$user"/jd/JDownloader.jar
```

Comment:
Do we need a whole `jar` for each user? why not use one `jar` and have all the services refer to that bin?

Reply:
b said something similar on the Discord. I will certainly be looking into this once everything else is said and done. I'm not 100% sure which files can be duplicated across installations. Although we would certainly need the `cfg` directory generated by each intallation to be separate. As there is an absolute plethora of options for JDownloader, and at the very minimum each users account information for accessing the web ui via the third party would have to be self contained.

It's quite possible that I could do something like drop the JDownloader.jar into /opt/jdownloader instead of to each user's home directory. The service file passes the working directory for JDownloader, so where the actual .jar is containted could be inconsequential.

However, I'm also not completely certain that that JDownloader.jar file will not be replaced at some point by an internal update. My concern here being that if one user's configuration was not updated along with the application files (for instance if they had jdownloader disabled for some time) then we might end up with some broken installations.

TL;DR - I'm not sure that having multiple copies of a 3.8 MB .jar file is something we should be worried about if it might cause more headaches down the line.

Reply:
Ait, let's keep this thread open until the end of the PR and let's revisit it then

Reply:
Upon further review, it would seem that JDownloader really doesn't care where JD_HOME is set to (during the intial runs at least). It seems to generate the files around itself, wherever it may be being run from.

Don't have high hopes of this being an option at this point.

Reply:
Just thought I'd give it a brief Google search.
[Doesn't sound promising.](https://board.jdownloader.org/showthread.php?t=74492)

Reply:
Fair, that pretty much says it's not possible and supported so I'm closing this one.